bikespace_frontend/src/components/Time.tsx

```diff
@@ -1,5 +1,79 @@
 import React from "react";
+import { ParkingDuration } from "../interfaces/Submission";
+import BaseButton from "./BaseButton";
 
-export default () => {
-  return <h1>Time</h1>;
+const Time = (props: { parkingDuration: ParkingDuration, onParkingDurationChanged: (parkingDuration: ParkingDuration) => void, dateTime: Date, onDateTimeChanged: (dateTime: Date) => void }) => {
+    const parkingDuration = props.parkingDuration;
+    const dateTime = props.dateTime;
+
+    const handleDateTime = (e: React.ChangeEvent<HTMLInputElement>) => {
+        e.preventDefault();
+        let chosenDateTime = new Date(e.currentTarget.value);
+        props.onDateTimeChanged(chosenDateTime);
+    };
+
+    const handleClick = (e: React.MouseEvent<HTMLButtonElement>) => {
+        e.preventDefault();
+        let duration: ParkingDuration;
+        switch (e.currentTarget.dataset.value) {
+            case "minutes":
+                duration = ParkingDuration.Minutes;
+                break;
+            case "hours":
+                duration = ParkingDuration.Hours;
+                break;
+            case "overnight":
+                duration = ParkingDuration.Overnight;
+                break;
+            case "multiday":
+                duration = ParkingDuration.MultiDay;
+                break;
+            default:
+                duration = ParkingDuration.Overnight;
+                break;
+        }
+        props.onParkingDurationChanged(duration);
+    };
+
+    return (
+        <div id="submission-time">
+            <h2>When did this happen?</h2>
+            <input value={dateTime.toISOString().slice(0,-8)} type="datetime-local" onChange={handleDateTime}>
```

Comment:
I think it'll use UTC time. I tried to assign a default time but it's ridiculously difficult to do that!

Reply:
I saw people use 2 libraries, moment and moment-timezone. It's bad, no wonder no one really like javascript 😂

Reply:
<img width="835" alt="image" src="https://github.com/bikespace/bikespace-v2/assets/1151907/d38a2fbc-ccec-4dc2-a676-09d78089eef3">